go/types/map_test.go

```diff
@@ -1155,3 +1155,41 @@ func TestMapRemoveLastWhenNotLoaded(t *testing.T) {
 		assert.True(tm.toMap().Equals(nm))
 	}
 }
+
+func TestMapIterFrom(t *testing.T) {
+	assert := assert.New(t)
+
+	test := func(m Map, rangeStart, rangeEnd, expectedStart, expectedEnd Value) {
+		first := true
+		stop := false
+		var end Value = Number(-1000)
+		m.IterFrom(rangeStart, func(k, v Value) bool {
+			assert.False(stop, "IterFrom callback called after returning true")
+			if first {
+				first = false
+				assert.True(k.Equals(expectedStart))
+			}
+			stop = rangeEnd != nil && rangeEnd.Less(k)
+			if !stop {
+				end = k
+			}
+			return stop
+		})
+		if rangeEnd == nil {
+			assert.False(stop, "nil expectedEnd but stop == true")
+		} else {
+			assert.True(end.Equals(expectedEnd), "end: %s != expectedEnd: %s", EncodedValue(end), EncodedValue(expectedEnd))
+		}
+	}
+
+	mapKVs := ValueSlice{}
+	for i := -98; i <= 98; i += 2 {
+		mapKVs = append(mapKVs, Number(i), Bool(true))
+	}
+
+	m1 := NewMap(mapKVs...)
```

Comment:
How about just `NewMap(generateNumbersAsValuesFromToBy(-98, 98, 1)...)`? Your map values will be numbers instead of `Bool` but that should be fine.